presto-spi/src/main/java/com/facebook/presto/spi/ConnectorTableLayout.java

```diff
@@ -123,6 +139,11 @@ public Optional<RowExpression> getRemainingPredicate()
         return remainingPredicate;
     }
 
+    public Optional<ColumnHandle> getUniqueColumn()
```

Comment:
Couldn't the uniqueness of the column be considered a LocalProperty (line 185 in this file)?

Reply:
Oh, I see the problem--even if there is ` [U($row_id), G(orderstatus)]`, it won't be right because it prevents further information about the properties of the data.  In the case of row ID, it will always be unique, in spite of the further properties, so to your point the unique property should be treated independently.

Reply:
> I'm also wondering if we should consider uniqueness as a special case of grouping, since if it is unique it is grouped?

Yeah, unique is a special case for grouping, one key difference which makes me creating a new property is unique is order insensitive but grouping is order sensitive, i.e. if output is reordered, grouping property may not hold, but uniqueness will continue hold.